typing_extensions/src_py3/typing_extensions.py

```diff
@@ -620,3 +665,269 @@ def new_type(x):
 else:
     # Constant that's True when type checking, but False here.
     TYPE_CHECKING = False
+
+
+def _collection_protocol(cls):
+    # Selected set of collections ABCs that are considered protocols.
+    name = cls.__name__
+    return (name in ('ABC', 'Callable', 'Awaitable',
+                     'Iterable', 'Iterator', 'AsyncIterable', 'AsyncIterator',
+                     'Hashable', 'Sized', 'Container', 'Collection', 'Reversible',
+                     'Sequence', 'MutableSequence', 'Mapping', 'MutableMapping',
+                     'AbstractContextManager', 'ContextManager',
+                     'AbstractAsyncContextManager', 'AsyncContextManager',) and
+            cls.__module__ in ('collections.abc', 'typing', 'contextlib',
+                               '_abcoll', 'abc'))
+
+
+class ProtocolMeta(GenericMeta):
```

Comment:
Probably yes, this makes sense not to expose it (I think we can't give guarantees about ``ProtocolMeta`` anyway). I will update now.